internal/document.go

```diff
@@ -0,0 +1,57 @@
+package internal
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Document represents the AWS Nitro Enclave Attestation Document.
+type Document struct {
+	ModuleID    string          `cbor:"module_id" json:"module_id"`
+	Timestamp   uint64          `cbor:"timestamp" json:"timestamp"`
+	Digest      string          `cbor:"digest" json:"digest"`
+	PCRs        map[uint][]byte `cbor:"pcrs" json:"pcrs"`
+	Certificate []byte          `cbor:"certificate" json:"certificate"`
+	CABundle    [][]byte        `cbor:"cabundle" json:"cabundle"`
+
+	PublicKey []byte `cbor:"public_key" json:"public_key,omitempty"`
+	UserData  []byte `cbor:"user_data" json:"user_data,omitempty"`
+	Nonce     []byte `cbor:"nonce" json:"nonce,omitempty"`
+}
+
+func (doc Document) CreatedAt() time.Time {
+	if doc.Timestamp == 0 {
+		return time.Time{}
+	}
+
+	// Pg. 64 of https://docs.aws.amazon.com/pdfs/enclaves/latest/user/enclaves-user.pdf
+	// describes Timestamp as "UTC time when document was created, in milliseconds".
+	// On the other, self-signed attestation timestamps are in seconds, so
+	// we need to figure out which time of timestamp to convert.
+	// TODO: Remove this check and use UnixMilli as part of
+	//  https://blocky.atlassian.net/browse/BKY-5620
+
+	var createdAt time.Time
+	if time.UnixMilli(int64(doc.Timestamp)).Year() < 1980 {
+		createdAt = time.Unix(int64(doc.Timestamp), 0)
+	} else {
+		createdAt = time.UnixMilli(int64(doc.Timestamp))
+	}
+
+	return createdAt
+}
+
+func (doc Document) Debug() (bool, error) {
+	pcr0, ok := doc.PCRs[0]
+	if !ok {
+		return false, fmt.Errorf("PCR0 not found")
+	}
+
+	pcr0Int, err := strconv.Atoi(hex.EncodeToString(pcr0))
```

Comment:
After looking at the implementation, I don't understand the benefit of doing the encoding and conversion instead of the "dumb" implementation of:
```
for x := range pcr0 {
    if pcr[0] != 0 {
        return false
    }
    return true
}
```

If it is an issue of clarity, at least to me, the "dumb" solution is more clear.  If it is an issue of "speed", I did a quick test with code:
```
package main_test

import (
	"encoding/hex"
	"strconv"
	"testing"
)

func mustDecode(s string) []byte {
	b, err := hex.DecodeString(s)
	if err != nil {
		panic(err)
	}

	return b
}

var (
	nonDebug = mustDecode("e72a46ca80a260fb044a125442f0c7e331813bcbaf9724d9f3857758992766f2d65710a27aa94ae3949dd54e7c9fe86a")
	debug    = mustDecode("000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000")
)

func isDebug1(v []byte) bool {
	for _, vi := range v {
		if vi != 0 {
			return false
		}
	}

	return true
}

func Benchmark1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		isDebug1(nonDebug)
		isDebug1(debug)
	}
}

func isDebug2(v []byte) bool {
	pcr0Int, err := strconv.Atoi(hex.EncodeToString(v))
	return err == nil && pcr0Int == 0
}

func Benchmark2(b *testing.B) {
	for i := 0; i < b.N; i++ {
		isDebug2(nonDebug)
		isDebug2(debug)
	}
}
```

Results:
```
goos: linux                                                                                                                                                                                                        
goarch: amd64                                                                                                                                                                                                      
pkg: bky.sh/dev-go                                                                                                                                                                                                 
cpu: Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz                                                                                                                                                                     
Benchmark1-8    87996946                14.63 ns/op                                                                                                                                                                
Benchmark2-8     2646614               471.3 ns/op                                                                                                                                                                 
```

The "dumb" solution is about 30x faster.  (Note that usually, I would not go to the trouble of doing a benchmark  ... I was looking for an excuse to try out a vim thing... lol)

Reply:
Reimplemented with a loop